### PR TITLE
Agreements: Use new ACL version

### PR DESCRIPTION
**Pull request by bingen**

### CODE REVIEW

File: apps/agreement/contracts/Agreement.sol [hunk 1]
```diff
@@ -1125,8 +1125,7 @@ contract Agreement is IAgreement, AragonApp {
             return false;
         }
 
-        // TODO: update with new ACL version: no need to pass challenger address by parameter
-        bytes memory params = ConversionHelpers.dangerouslyCastUintArrayToBytes(arr(_challenger));
+        bytes memory params = new bytes(0);
```
Comment (facuspagnuolo) [inline apps/agreement/contracts/Agreement.sol:1128]: We can probably inline this one now :)

File: apps/agreement/contracts/test/mocks/disputable/DisputableAppMock.sol [hunk 2]
```diff
@@ -91,7 +91,7 @@ contract DisputableAppMock is DisputableAragonApp, TimeHelpersMock {
     * @return True if the given address can submit actions, false otherwise
     */
     function canForward(address _sender, bytes) public view returns (bool) {
-        return canPerform(_sender, SUBMIT_ROLE, arr(_sender));
+        return canPerform(_sender, SUBMIT_ROLE, new uint256[](0));
```
Comment (facuspagnuolo) [inline apps/agreement/contracts/test/mocks/disputable/DisputableAppMock.sol:94]:
```suggestion
        return canPerform(_sender, SUBMIT_ROLE, arr());
```

File: apps/agreement/contracts/test/mocks/helpers/TokenBalanceOracle.sol [hunk 3]
```diff
@@ -22,14 +22,8 @@ contract TokenBalanceOracle is IACLOracle {
...
-    function canPerform(address, address, address, bytes32, uint256[] _how) external view returns (bool) {
-        require(_how.length > 0, ERROR_SENDER_MISSING);
-        require(_how[0] < 2**160, ERROR_SENDER_TOO_BIG);
-        require(_how[0] != 0, ERROR_SENDER_ZERO);
-
-        address sender = address(_how[0]);
-
-        uint256 senderBalance = token.balanceOf(sender);
+    function canPerform(address _who, address, address, bytes32, uint256[]) external view returns (bool) {
+        uint256 senderBalance = token.balanceOf(_who);
```
Comment (facuspagnuolo) [inline apps/agreement/contracts/test/mocks/helpers/TokenBalanceOracle.sol:26]: 🙌

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 4]
```diff
@@ -499,7 +499,7 @@ contract DisputableVoting is DisputableAragonApp, IForwarder {
     function canForward(address _sender, bytes) public view returns (bool) {
         // TODO: Handle the case where a Disputable app doesn't have an Agreement set
         // Note that `canPerform()` implicitly does an initialization check itself
-        return canPerform(_sender, CREATE_VOTES_ROLE, arr(_sender));
+        return canPerform(_sender, CREATE_VOTES_ROLE, new uint256[](0));
```
Comment (facuspagnuolo) [inline apps/voting-disputable/contracts/DisputableVoting.sol:502]:
```suggestion
        return canPerform(_sender, CREATE_VOTES_ROLE, arr());
```

File: apps/agreement/contracts/Agreement.sol (hunk 1 above)
Comment (bingen) [reply apps/agreement/contracts/Agreement.sol:1128]: Sure! I was about to do it, but I thought you liked this style :sweat_smile:

**Comment by bingen** [comment]
As suggested, I split this into #1209 and #1210, and merged the last one.

**Pull request closed without merging**